scilpy/utils/streamlines.py

```diff
@@ -93,30 +94,82 @@ def uniformize_bundle_sft(sft, axis=None, swap=False):
          The tractogram that contains the list of streamlines to be uniformized
     axis: int, optional
         Orient endpoints in the given axis
+    ref_bundle: streamlines
+        Orient endpoints the same way as this bundle (or centroid)
     swap: boolean, optional
         Swap the orientation of streamlines
-
     """
+    from scilpy.tractanalysis.reproducibility_measures \
+        import get_endpoints_density_map
+    old_space = sft.space
+    old_origin = sft.origin
+    sft.to_vox()
+    sft.to_corner()
+    density = get_endpoints_density_map(sft.streamlines, sft.dimensions,
+                                        point_to_select=3)
+    indices = np.argwhere(density > 0)
+    kmeans = KMeans(n_clusters=2, random_state=0, copy_x=True,
+                    precompute_distances=True, n_init=20,
+                    n_jobs=1).fit(indices)
+
+    labels = np.zeros(density.shape)
+    for i in range(len(kmeans.labels_)):
+        labels[tuple(indices[i])] = kmeans.labels_[i]+1
+
+    k_means_centers = kmeans.cluster_centers_
+    main_dir_barycenter = np.argmax(
+        np.abs(k_means_centers[0] - k_means_centers[-1]))
+
     if len(sft.streamlines) > 0:
         axis_name = ['x', 'y', 'z']
-        if axis is None:
-            centroid = get_streamlines_centroid(sft.streamlines, 20)[0]
+        if axis is None or ref_bundle is not None:
+            if ref_bundle is not None:
+                ref_bundle.to_vox()
+                ref_bundle.to_corner()
+                centroid = get_streamlines_centroid(ref_bundle.streamlines,
+                                                    20)[0]
+            else:
+                centroid = get_streamlines_centroid(sft.streamlines, 20)[0]
```

Comment:
Why 20 points has never been an arg and is hardcoded ?

Reply:
This is simply because I have to convert streamlines to an array with an equal number of pts, it could be 10, 20, 50, but in the end, it changes virtually nothing.